src/main/java/fi/hsl/transitdata/pulsarpubtransconnect/PubtransTableHandler.java

```diff
@@ -102,21 +136,28 @@ public Collection<TypedMessageBuilder<byte[]>> handleResultSet(ResultSet resultS
                     metroTripCount++;
                     metroRouteIds.add(tripInfo.getRouteId());
                 } else {
-                    final byte[] data = createPayload(resultSet, common, tripInfo);
-                    TypedMessageBuilder<byte[]> msgBuilder = createMessage(key, eventTimestampUtcMs, dvjId, data, getSchema());
+                    final byte[] data = createPayload(queryResultItem.common, queryResultItem.columnToIdMap, tripInfo);
+                    TypedMessageBuilder<byte[]> msgBuilder = createMessage(queryResultItem.key,
+                            queryResultItem.eventTimestampUtcMs, dvjId, data, getSchema());
                     messageBuilderQueue.add(msgBuilder);
                 }
             }
 
             //Update latest ts for next round
-            if (eventTimestampUtcMs > tempTimeStamp) {
-                tempTimeStamp = eventTimestampUtcMs;
+            if (queryResultItem.eventTimestampUtcMs > tempTimeStamp) {
+                tempTimeStamp = queryResultItem.eventTimestampUtcMs;
             }
         }
-
-        log.info("{} rows processed from the result set. {} rows skipped with metro trips (route ids: {})",
-                count, metroTripCount, metroRouteIds);
-
+        
+        long endTime = System.currentTimeMillis();
+        resultHandlerDuration = endTime - queryEndTime;
+        queryAndResultHandlerDuration = endTime - queryStartTime;
+        
+        log.info("{} rows processed from the result set. {} rows skipped with metro trips (route ids: {}). "
+                        + "Operation took {} (db query took {}, handling results took {})",
+                count, metroTripCount, metroRouteIds,
+                getMinSec(queryAndResultHandlerDuration), getMinSec(queryDuration), getMinSec(resultHandlerDuration));
```

Comment:
While pretty, I would avoid making the log message more human-readable in this way. When we switch to structured logging, the variable for time here should contain for example the seconds as float. That way the variable can be used for alerts directly.